service/storage_service.cc

```diff
@@ -1637,9 +1643,39 @@ class topology_coordinator {
     }
 
     future<> generate_migration_updates(std::vector<canonical_mutation>& out, const group0_guard& guard, const migration_plan& plan) {
+        std::unordered_set<table_id> new_transitions;
         for (const tablet_migration_info& mig : plan.migrations()) {
             co_await coroutine::maybe_yield();
             generate_migration_update(out, guard, mig);
+            new_transitions.insert(mig.tablet.table);
+        }
+
+        for (auto [table_id, resize_decision] : plan.resize_plan().resize) {
+            auto s = _db.find_schema(table_id);
+            resize_decision.phase = guard.write_timestamp();
+            slogger.debug("Generating resize decision {} with phase {} for table {}", resize_decision.name(), resize_decision.phase, table_id);
+            out.emplace_back(
+                replica::tablet_mutation_builder(guard.write_timestamp(), s->ks_name(), table_id)
+                    .set_resize_decision(std::move(resize_decision))
+                    .build());
+        }
+
+        // FIXME: Finalize split requests when exiting the tablet migration track.
+        for (auto table_id : plan.resize_plan().finalize_resize) {
+            auto& tmap = get_token_metadata_ptr()->tablets().get_tablet_map(table_id);
+            // Only finalize split request if there's no ongoing transition or new ones for a given table.
+            if (tmap.transitions().size() > 0 || new_transitions.contains(table_id)) {
+                continue;
+            }
+
+            auto s = _db.find_schema(table_id);
+            auto new_tablet_map = co_await _tablet_allocator.split_tablets(get_token_metadata_ptr(), table_id);
```

Comment:
So, there is no moving (migration) of tablets yet. Only split. After the split. The load balancing will do the real migration in another plan?

Do we use any of the tablet_transition_stage for the resize?

Reply:
@asias exactly. usually intra-node migration is needed after split, such that new tablets can leave the original shards, but still being worked on by Tomek. as for resize, the plan is that state machine jumps into a new transition stage for split after leaving migration track.